### DISCUSSION TITLE
Consider generating Markdown + Mermaid flowchart graph as output type

### DISCUSSION THREAD

**Discussion by nicop4** [madushans/DependenSee#33] [Ideas]
Hi

I use Dependensee to visualize complex solutions and I manually transform the json graphviz output to a mardown document with a mermaid graph with nice results with a [mermaid flowchart](https://mermaid.js.org/syntax/flowchart.html). 

![image](https://user-images.githubusercontent.com/10168765/237042036-78075674-a69d-4cf1-952b-8e5ef48e4844.png)

I use markdown a lot and a lot of editors have extensive compatibility with it.

Did you consider generating markdown output ?

**Comment by madushans** [comment]
hi,

My plan going forward is to externalize the visualization stuff except for the HTML output. TL;DR; is you can use the JSON/XML output types and transforming the output into the format you want, but that does involve writing some code/script to do so.

---

In the original version, the idea was to allow new formats natively. This is what resulted in the GraphViz output type. But going forward I'm thinking it may be better to make the object model as the "interface", and allow external tools to do the custom visualizations.

This avoids DependenSee from having to know a lot of formats where most people only care about the HTML visualization. Also some formats may not be able to visualize them correctly when some nodes were excluded via filtering parameters, causing the tree to have multiple disconnected "islands".

I'm working on the next version which makes the html canvas visualization the primary/only output, and publishing a [separate nuget package](https://github.com/madushans/DependenSee/issues/28) that can provide a C# object tree to the calling code and let it handle the visualization. (And may be also keep the JSON output, if someone wants to do it externally without having to write C# or powershell)

Downside of this is that the nuget package won't allow filtering, and that responsibility is given to the visualization code.

If you're keen on this approach, you can try the Json or XML output, then use your own code to transform to a mermaid chart. (Current version will still use the filtering you specify at the command line)

I haven't had much time to work on the v3 work, so I don't really have a "target release date" for this stuff 😄 but the [v3 branch is public if you want to have a peek of what's planned.
